conans/client/tools/net.py

```diff
@@ -9,10 +9,13 @@
 _global_requester = None
 
 
-def get(url, md5='', sha1='', sha256='', destination="."):
+def get(url, md5='', sha1='', sha256='', destination=".", filename=""):
     """ high level downloader + unzipper + (optional hash checker) + delete temporary zip
     """
-    filename = os.path.basename(url)
+    if not filename and ("?" in url or "=" in url):
+        raise ConanException("Cannot deduce file name form url. Use 'filename' parameter.")
```

Comment:
I have a concern here. Is it really necessary that the user specifies a ``filename``? After all, the downloaded file is removed at the end of the method, so it is a temporary name, could probably just be some arbitrary name.

cc/ @lasote

Reply:
AFAIK it can be a random name because the unzip() needs the extension of the file to decompress, otherwise it will consider all files as ZIP by default